core/src/data/filters.h

```diff
@@ -0,0 +1,172 @@
+#pragma once
+
+#include "tileData.h"
+#include <unordered_map>
+#include <vector>
+#include <limits>
+
+namespace Tangram {
+
+    struct Value {
+
+        float num;
+        std::string str;
+
+        virtual bool equals(float f) const = 0;
+        virtual bool equals(const std::string& s) const = 0;
+        virtual bool equals(const Value& v) const = 0;
+
+        virtual ~Value() {};
+
+        Value(float n) : num(n) {}
+        Value(float n, const std::string& s) : num(n), str(s) {}
+        Value(const std::string& s) : str(s) {}
+
+    };
+
+    struct NumValue : public Value {
+
+        NumValue(float n) : Value(n) {}
+        NumValue(float n, const std::string& s) : Value(n, s) {}
+        ~NumValue() {}
+
+        virtual bool equals(const std::string& s) const override { return str.size() != 0 && str == s; }
+        virtual bool equals(float f) const override { return num == f; }
+        virtual bool equals(const Value& v) const override { return v.equals(num); }
+
+    };
+
+    struct StrValue : public Value {
+
+        StrValue(const std::string& s) : Value(s) {}
+        ~StrValue() {}
+
+        virtual bool equals(const std::string& s) const override { return str == s; }
+        virtual bool equals(float f) const override { return false; }
+        virtual bool equals(const Value& v) const override { return v.equals(str); }
+
+    };
+
+    using ValueList = std::vector<Value*>;
+
+    using Context = std::unordered_map<std::string, Value*>;
+
+    struct Filter {
+
+        virtual bool eval(const Feature& f, const Context& c) const { return false; };
```

Comment:
True.

So something like this should pass!! But it will not in the present implementation.

'''
YAML::Node node = YAML::Load("filter: ");
Filter* filter = sceneLoader.generateFilter(node["filter"]);

REQUIRE(filter->eval(civic, ctx));
'''